DigitalLearningSolutions.Web/Controllers/LearningPortalController/RecommendedLearningController.cs

```diff
@@ -59,20 +60,39 @@ public async Task<IActionResult> SelfAssessmentResults(int selfAssessmentId)
             return RedirectToAction("RecommendedLearning", new { selfAssessmentId });
         }
 
-        [Route("/LearningPortal/SelfAssessment/{selfAssessmentId:int}/RecommendedLearning")]
-        public async Task<IActionResult> RecommendedLearning(int selfAssessmentId)
+        [Route("/LearningPortal/SelfAssessment/{selfAssessmentId:int}/RecommendedLearning/{page:int=1}")]
+        public async Task<IActionResult> RecommendedLearning(
+            int selfAssessmentId,
+            int page = 1,
+            string? searchString = null
+        )
         {
             if (!configuration.IsSignpostingUsed())
             {
                 return RedirectToAction("FilteredDashboard", new { selfAssessmentId });
             }
 
             var candidateId = User.GetCandidateIdKnownNotNull();
-            string destUrl = "/LearningPortal/SelfAssessment/" + selfAssessmentId + "/RecommendedLearning";
+            var destUrl = "/LearningPortal/SelfAssessment/" + selfAssessmentId + "/RecommendedLearning";
             selfAssessmentService.SetBookmark(selfAssessmentId, candidateId, destUrl);
             selfAssessmentService.UpdateLastAccessed(selfAssessmentId, candidateId);
 
-            return await ReturnSignpostingRecommendedLearningView(selfAssessmentId, candidateId);
+            return await ReturnSignpostingRecommendedLearningView(selfAssessmentId, candidateId, page, searchString);
+        }
+
+        [Route("/LearningPortal/SelfAssessment/{selfAssessmentId:int}/AllRecommendedLearningItems")]
+        public async Task<IActionResult> AllRecommendedLearningItems(int selfAssessmentId)

```

Comment:
Should we perhaps be using the FeatureGate attribute to prevent accessing this endpoint when Signposting is not enabled?

Reply:
Good idea, I've applied the FeatureGate to that action which made it a bit neater as well.